httpbin/middleware.go

```diff
@@ -92,10 +103,11 @@ func (mw *metaResponseWriter) Size() int {
 
 func logger(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		reqMethod, reqURI := r.Method, r.URL.RequestURI()
```

Comment:
Capturing `reqMethod` is necessary here because our new `metaRequests` middleware will modify the method for `HEAD` requests, right?

Reply:
Yes, but also because something else could conceivably modify the method or URL in the downstream handlers as well, if not now then in the future. It seemed simpler and more future-proof for the `logger` to save the method and URI just in case, and it's much more efficient than cloning the whole request in the downstream headers.

Reply:
Cool, makes sense!